aten/src/ATen/core/PythonFallbackKernel.cpp

```diff
@@ -6,23 +6,56 @@
 
 namespace {
 
-// TLS saving the state of the include/exclude sets on entry to the dispatcher
-// This is set in the pythonTLSSnapshot fallback and used by the Python fallback.
-thread_local std::stack<c10::impl::LocalDispatchKeySet> tls_on_entry;
+// This TLS is used to track the state of the dispatcher to be able to restore
+// it when calling back into python.
+// It has the following invariant:
+//  - It must be empty while python code is executed.
+//  - It should only be set once even for multiple dispatcher calls that do not come
```

Comment:
and specifically, it gets set by the first dispatcher call, and ignored by any subsequent dispatcher calls? (until you eventually get to the python fallback and clear it).